cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -344,14 +344,18 @@ public void startAdapter(Id.Namespace namespace, String adapterName)
     setAdapterStatus(namespace, adapterName, AdapterStatus.STARTED);
   }
 
+  private Id.Program getWorkflowId(Id.Namespace namespace, AdapterSpecification adapterSpec) {
+    ApplicationSpecification appSpec = store.getApplication(Id.Application.from(namespace, adapterSpec.getTemplate()));
+    Preconditions.checkNotNull(appSpec);
+    Preconditions.checkArgument(appSpec.getWorkflows().size() == 1);
+    String workflowName = Lists.newArrayList(appSpec.getWorkflows().keySet()).get(0);
```

Comment:
Why need to new an array list? Use `Iterables.getFirst(appSpec.getWorkflows().keySet())`.